test/e2e/api_manager.go

```diff
@@ -0,0 +1,369 @@
+package e2e
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/rhobs/rhobs-synthetics-agent/internal/api"
+)
+
+// RealAPIManager manages the lifecycle of the actual RHOBS Synthetics API server
+type RealAPIManager struct {
+	cmd         *exec.Cmd
+	apiURL      string
+	port        int
+	dataDir     string
+	stopChan    chan struct{}
+	started     bool
+	ctx         context.Context
+	cancel      context.CancelFunc
+}
+
+// NewRealAPIManager creates a new manager for the real API server
+func NewRealAPIManager() *RealAPIManager {
+	ctx, cancel := context.WithCancel(context.Background())
+	
+	return &RealAPIManager{
+		port:     8080,
+		dataDir:  "/tmp/rhobs-synthetics-api-test-data",
+		stopChan: make(chan struct{}),
+		ctx:      ctx,
+		cancel:   cancel,
+	}
+}
+
+// Start builds and starts the real RHOBS Synthetics API server
+func (m *RealAPIManager) Start() error {
+	if m.started {
+		return fmt.Errorf("API server already started")
+	}
+
+	// Build the API server first
+	if err := m.buildAPI(); err != nil {
+		return fmt.Errorf("failed to build API: %w", err)
+	}
+
+	// Create data directory
+	if err := os.MkdirAll(m.dataDir, 0755); err != nil {
+		return fmt.Errorf("failed to create data directory: %w", err)
+	}
+
+	// Find an available port
+	if err := m.findAvailablePort(); err != nil {
+		return fmt.Errorf("failed to find available port: %w", err)
+	}
+
+	m.apiURL = fmt.Sprintf("http://localhost:%d", m.port)
+
+	// Start the API server
+	if err := m.startAPI(); err != nil {
+		return fmt.Errorf("failed to start API: %w", err)
+	}
+
+	// Wait for API to be ready
+	if err := m.waitForAPI(); err != nil {
+		_ = m.Stop()
+		return fmt.Errorf("API failed to become ready: %w", err)
+	}
+
+	// Seed with test data
+	if err := m.SeedTestData(); err != nil {
+		_ = m.Stop()
+		return fmt.Errorf("failed to seed test data: %w", err)
+	}
+
+	m.started = true
+	return nil
+}
+
+// Stop shuts down the API server
+func (m *RealAPIManager) Stop() error {
+	if !m.started {
+		return nil
+	}
+
+	m.cancel()
+	close(m.stopChan)
+
+	if m.cmd != nil && m.cmd.Process != nil {
+		// Try graceful shutdown first
+		if err := m.cmd.Process.Signal(syscall.SIGTERM); err != nil {
+			// Force kill if graceful shutdown fails
+			_ = m.cmd.Process.Kill()
+		}
+		_ = m.cmd.Wait()
+	}
+
+	// Clean up data directory
+	_ = os.RemoveAll(m.dataDir)
+
+	m.started = false
+	return nil
+}
+
+// GetURL returns the API server URL
+func (m *RealAPIManager) GetURL() string {
+	return m.apiURL
+}
+
+// buildAPI builds the RHOBS Synthetics API binary
+func (m *RealAPIManager) buildAPI() error {
+	apiPath := "/Users/drow/src/rhobs-synthetics-api"
```

Comment:
Probably need to figure out how to make this portable for a CI system

Reply:
yep i fixed this